src/App/App.jsx

```diff
@@ -1,21 +1,69 @@
-import { Switch, Route } from 'react-router-dom';
-import { Home } from '../components/Home/Home';
+import React from 'react';
+import { Switch, Route, Redirect } from 'react-router-dom';
+import { Home } from '../components/Pages/Home/Home';
 import { Members } from '../components/Pages/Members/Members';
 import { Tasks } from '../components/Pages/Tasks/Tasks';
+import { AllTasks } from '../components/Pages/Tasks/AllTasks';
 import { Progress } from '../components/Pages/Progress/Progress';
-import style from './App.module.css';
+import { Header } from '../components/Common/Header/Header';
+import { Footer } from '../components/Common/Footer/Footer';
+import { Login } from '../components/Pages/Login/Login';
+import { Track } from '../components/Pages/Track/Track';
 
-export const App = () => {
-  return (
-    <div className={style.content}>
-      <Switch>
-        <Route exact path='/'>
-          <Home />
-        </Route>
-        <Route path='/members' component={Members} />
-        <Route path='/progress/:id' component={Progress} />
-        <Route path='/tasks/:id' component={Tasks} />
-      </Switch>
-    </div>
-  );
-};
+export class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAuth: false,
+      userEmail: '',
+    };
+  }
+
+  setAuth = (userEmail) => {
+    if (userEmail) {
+      this.setState({ userEmail, isAuth: true });
+    }
+  };
+
+  logout = () => {
+    this.setState({ userEmail: '', isAuth: false });
+  };
+
+  render() {
+    const { isAuth, userEmail } = this.state;
+
+    return (
+      <>
+        <Header userEmail={userEmail} handleLogout={this.logout} />
+        <main>
+          <Switch>
+            <Route exact path='/' component={Home} />
+            <Route path='/members' render={() => (!isAuth ? <Redirect to='/login' /> : <Members />)} />
+            <Route
+              exact
+              path='/progress/:id'
+              render={(params) => (!isAuth ? <Redirect to='/login' /> : <Progress params={params} />)}
+            />
+            <Route exact path='/tasks' render={() => (!isAuth ? <Redirect to='/login' /> : <AllTasks />)} />
+            <Route
+              exact
+              path='/tasks/:id'
+              render={(params) => (!isAuth ? <Redirect to='/login' /> : <Tasks params={params} />)}
+            />
+            <Route
+              exact
+              path='/tasks/track/:id'
+              render={(params) => (!isAuth ? <Redirect to='/login' /> : <Track params={params} />)}
+            />
+            <Route
+              exact
+              path='/login'
+              render={() => <Login handleAuth={this.setAuth} isAuth={isAuth} userEmail={userEmail} />}
```

Comment:
use component instead of render

Reply:
```
    <Switch>
      <Route exact path='/' component={Home} />
      <Route exact path='/progress/:id' component={Progress} />
      <Route exact path='/members' component={Members} />
      <Route exact path='/tasks' component={AllTasks} />
      <Route exact path='/tasks/:id' component={AllTasks} />
      <Route exact path='/tasks/:id/tracks' render={(params) => <Tracks params={params} userId={userId} />} />
    </Switch>
```
If i need parameters inside component how can i do without render?